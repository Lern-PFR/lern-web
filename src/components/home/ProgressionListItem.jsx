import PropTypes from 'prop-types';
import { StyledListItem } from 'components/shared/styledElements';
import { BodyCopy, GreatPrimer } from 'components/shared/typography';
import { progressionListItem } from 'theme/pages/home/homepageAuth';
import { getModuleById } from 'redux/selectors/modules';

/**
 * @name ProgressionListItem
 * @description A custom list item to display an instance of a user's progression in a subject
 *
 * @author Christopher Walker
 *
 * @param {string} updatedAt		The last time this progression was updated - the last time the user advanced in this subject.
 * @param {object} subject			An instance of the subject that progression is being shown for.
 * @param {object} latestConcept	An instance of the last concept that the user completed in this subject.
 * @param {number} completion		The percentage of the subject that the user has completed.
 */
const ProgressionListItem = ({ updatedAt, subject, latestConcept, completion }) => {
	const module = getModuleById(latestConcept.moduleId);

	return (
		<StyledListItem {...progressionListItem}>
			<GreatPrimer margin="0">{updatedAt}</GreatPrimer>
			<BodyCopy margin="0">
				{subject.title}
				(
				{completion}
				)
			</BodyCopy>
			<BodyCopy margin="1px">{module.title}</BodyCopy>
			<BodyCopy margin="2px">{latestConcept.title}</BodyCopy>
		</StyledListItem>
	);
};

ProgressionListItem.propTypes = {
	updatedAt: PropTypes.string.isRequired,
	subject: PropTypes.shape({
		id: PropTypes.oneOfType([
			PropTypes.string,
			PropTypes.number,
		]).isRequired,
		title: PropTypes.string.isRequired,
	}).isRequired,
	latestConcept: PropTypes.shape({
		id: PropTypes.oneOfType([
			PropTypes.string,
			PropTypes.number,
		]).isRequired,
		title: PropTypes.string.isRequired,
		moduleId: PropTypes.oneOfType([
			PropTypes.string,
			PropTypes.number,
		]).isRequired,
	}).isRequired,
	completion: PropTypes.number.isRequired,
};

export default ProgressionListItem;
