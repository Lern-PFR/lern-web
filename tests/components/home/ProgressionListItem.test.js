import { mount } from 'enzyme';
import { Provider } from 'react-redux';
import configureMockStore from 'redux-mock-store';
import thunk from 'redux-thunk';

import ProgressionListItem from 'components/home/ProgressionListItem';

describe('ProgressionListItem', () => {
	const mockStore = configureMockStore([thunk]);

	afterEach(() => {
		jest.restoreAllMocks();
		jest.resetAllMocks();
	});

	it('should match previous snapshot for progression within a single module', () => {
		const store = mockStore({
			modules: [
				{
					id: 'abcd',
					title: 'dummy module',
					description: 'dummy description',
					subjectId: 'zxcv',
					concepts: [
						{
							id: 'qwer',
							title: 'dummy concept',
							moduleId: 'abcd',
							order: 0,
						},
						{
							id: 'asdf',
							title: 'dummy concept',
							moduleId: 'abcd',
							order: 1,
						},
					],
				},
			],
		});
		const progression = {
			subject: {
				id: 'zxcv',
				title: 'dummy subject',
				description: 'dummy description',
				modules: [
					{
						id: 'abcd',
						title: 'dummy module',
						description: 'dummy description',
						subjectId: 'zxcv',
						concepts: [
							{
								id: 'qwer',
								title: 'dummy concept',
								moduleId: 'abcd',
								order: 0,
							},
							{
								id: 'asdf',
								title: 'dummy concept',
								moduleId: 'abcd',
								order: 1,
							},
						],
					},
				],
			},
			concept: {
				id: 'qwer',
				title: 'dummy concept',
				moduleId: 'abcd',
			},
			completion: 67,
			updatedAt: '2021-06-20T09:33:20.626274',
		};

		const sut = (
			<Provider store={store}>
				<ProgressionListItem {...progression} />
			</Provider>
		);
		const wrapper = mount(sut);

		expect(wrapper).toMatchSnapshot();
	});

	it('should match previous snapshot for progression spanning two modules', () => {
		const store = mockStore({
			modules: [
				{
					id: 'abcd',
					title: 'dummy module',
					description: 'dummy description',
					subjectId: 'zxcv',
					concepts: [
						{
							id: 'qwer',
							title: 'dummy concept',
							moduleId: 'abcd',
							order: 0,
						},
						{
							id: 'asdf',
							title: 'dummy concept',
							moduleId: 'abcd',
							order: 1,
						},
					],
				},
				{
					id: 'efgh',
					title: 'dummy module',
					description: 'dummy description',
					subjectId: 'zxcv',
					concepts: [
						{
							id: 'ghjk',
							title: 'dummy concept',
							moduleId: 'efgh',
							order: 0,
						},
						{
							id: 'yuio',
							title: 'dummy concept',
							moduleId: 'efgh',
							order: 1,
						},
					],
				},
			],
		});
		const progression = {
			subject: {
				id: 'zxcv',
				title: 'dummy subject',
				description: 'dummy description',
				modules: [
					{
						id: 'abcd',
						title: 'dummy module',
						description: 'dummy description',
						subjectId: 'zxcv',
						concepts: [
							{
								id: 'qwer',
								title: 'dummy concept',
								moduleId: 'abcd',
								order: 0,
							},
							{
								id: 'asdf',
								title: 'dummy concept',
								moduleId: 'abcd',
								order: 1,
							},
						],
					},
					{
						id: 'efgh',
						title: 'dummy module',
						description: 'dummy description',
						subjectId: 'zxcv',
						concepts: [
							{
								id: 'ghjk',
								title: 'dummy concept',
								moduleId: 'efgh',
								order: 0,
							},
							{
								id: 'yuio',
								title: 'dummy concept',
								moduleId: 'efgh',
								order: 1,
							},
						],
					},
				],
			},
			concept: {
				id: 'asdf',
				title: 'dummy concept',
				moduleId: 'abcd',
			},
			completion: 67,
			updatedAt: '2021-06-20T09:33:20.626274',
		};

		const sut = (
			<Provider store={store}>
				<ProgressionListItem {...progression} />
			</Provider>
		);
		const wrapper = mount(sut);

		expect(wrapper).toMatchSnapshot();
	});

	it('should match previous snapshot for progression within the last concept of the last module', () => {
		const store = mockStore({
			modules: [
				{
					id: 'abcd',
					title: 'dummy module',
					description: 'dummy description',
					subjectId: 'zxcv',
					concepts: [
						{
							id: 'qwer',
							title: 'dummy concept',
							moduleId: 'abcd',
							order: 0,
						},
						{
							id: 'asdf',
							title: 'dummy concept',
							moduleId: 'abcd',
							order: 1,
						},
					],
				},
			],
		});
		const progression = {
			subject: {
				id: 'zxcv',
				title: 'dummy subject',
				description: 'dummy description',
				modules: [
					{
						id: 'abcd',
						title: 'dummy module',
						description: 'dummy description',
						subjectId: 'zxcv',
						concepts: [
							{
								id: 'qwer',
								title: 'dummy concept',
								moduleId: 'abcd',
								order: 0,
							},
							{
								id: 'asdf',
								title: 'dummy concept',
								moduleId: 'abcd',
								order: 1,
							},
						],
					},
				],
			},
			concept: {
				id: 'asdf',
				title: 'dummy concept',
				moduleId: 'abcd',
			},
			completion: 67,
			updatedAt: '2021-06-20T09:33:20.626274',
		};

		const sut = (
			<Provider store={store}>
				<ProgressionListItem {...progression} />
			</Provider>
		);
		const wrapper = mount(sut);

		expect(wrapper).toMatchSnapshot();
	});
});
