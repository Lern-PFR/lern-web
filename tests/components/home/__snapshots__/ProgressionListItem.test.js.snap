// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProgressionListItem should match previous snapshot for progression spanning two modules 1`] = `
<Provider
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <ProgressionListItem
    completion={67}
    concept={
      Object {
        "id": "asdf",
        "moduleId": "abcd",
        "title": "dummy concept",
      }
    }
    subject={
      Object {
        "description": "dummy description",
        "id": "zxcv",
        "modules": Array [
          Object {
            "concepts": Array [
              Object {
                "id": "qwer",
                "moduleId": "abcd",
                "order": 0,
                "title": "dummy concept",
              },
              Object {
                "id": "asdf",
                "moduleId": "abcd",
                "order": 1,
                "title": "dummy concept",
              },
            ],
            "description": "dummy description",
            "id": "abcd",
            "subjectId": "zxcv",
            "title": "dummy module",
          },
          Object {
            "concepts": Array [
              Object {
                "id": "ghjk",
                "moduleId": "efgh",
                "order": 0,
                "title": "dummy concept",
              },
              Object {
                "id": "yuio",
                "moduleId": "efgh",
                "order": 1,
                "title": "dummy concept",
              },
            ],
            "description": "dummy description",
            "id": "efgh",
            "subjectId": "zxcv",
            "title": "dummy module",
          },
        ],
        "title": "dummy subject",
      }
    }
    updatedAt="2021-06-20T09:33:20.626274"
  >
    <styled.li
      borderLeft="1px solid #241C15"
      cursor="pointer"
      onClick={[Function]}
      paddingLeft="1.5em"
    >
      <li
        className="sc-hKgILt hnwkoi"
        cursor="pointer"
        onClick={[Function]}
      >
        <styled.div
          alignItems="center"
          cursor="pointer"
          display="flex"
          flexDirection="row"
        >
          <div
            className="sc-bdfBwQ gAYIqo"
            cursor="pointer"
            display="flex"
          >
            <styled.div
              bg="#241C15"
              border="1px solid #241C15"
              borderRadius="50%"
              cursor="pointer"
              height="13px"
              marginLeft="-31px"
              width="13px"
            >
              <div
                className="sc-bdfBwQ iTpKPV"
                cursor="pointer"
                height="13px"
                width="13px"
              />
            </styled.div>
            <styled.div
              cursor="pointer"
              marginLeft="1em"
              width="266px"
            >
              <div
                className="sc-bdfBwQ jSOydB"
                cursor="pointer"
                width="266px"
              >
                <GreatPrimer
                  cursor="pointer"
                  lineHeight="initial"
                  margin={0}
                >
                  <DynamicTextComponent
                    cursor="pointer"
                    fontFamily="IBM Plex Sans, Helvetica Neue, Arial, sans-serif"
                    fontSize="1.222rem"
                    fontWeight={400}
                    letterSpacing={0}
                    lineHeight="initial"
                    margin={0}
                    tag="p"
                  >
                    <styled.p
                      as="p"
                      cursor="pointer"
                      fontFamily="IBM Plex Sans, Helvetica Neue, Arial, sans-serif"
                      fontSize="1.222rem"
                      fontWeight={400}
                      letterSpacing={0}
                      lineHeight="initial"
                      margin={0}
                    >
                      <p
                        className="sc-gKsewC ebSZFE"
                        cursor="pointer"
                        fontFamily="IBM Plex Sans, Helvetica Neue, Arial, sans-serif"
                        fontSize="1.222rem"
                        fontWeight={400}
                        letterSpacing={0}
                      >
                        home.pages.auth.date
                      </p>
                    </styled.p>
                  </DynamicTextComponent>
                </GreatPrimer>
                <BodyCopy
                  cursor="pointer"
                  lineHeight="initial"
                  margin={0}
                >
                  <DynamicTextComponent
                    cursor="pointer"
                    fontFamily="IBM Plex Sans, Helvetica Neue, Arial, sans-serif"
                    fontSize="1rem"
                    fontWeight={400}
                    letterSpacing={0}
                    lineHeight="initial"
                    margin={0}
                    tag="p"
                  >
                    <styled.p
                      as="p"
                      cursor="pointer"
                      fontFamily="IBM Plex Sans, Helvetica Neue, Arial, sans-serif"
                      fontSize="1rem"
                      fontWeight={400}
                      letterSpacing={0}
                      lineHeight="initial"
                      margin={0}
                    >
                      <p
                        className="sc-gKsewC cIUUkE"
                        cursor="pointer"
                        fontFamily="IBM Plex Sans, Helvetica Neue, Arial, sans-serif"
                        fontSize="1rem"
                        fontWeight={400}
                        letterSpacing={0}
                      >
                        home.pages.auth.subject_completion
                      </p>
                    </styled.p>
                  </DynamicTextComponent>
                </BodyCopy>
              </div>
            </styled.div>
          </div>
        </styled.div>
        <styled.div
          alignItems="center"
          cursor="pointer"
          display="flex"
          flexDirection="row"
        >
          <div
            className="sc-bdfBwQ gAYIqo"
            cursor="pointer"
            display="flex"
          >
            <styled.div
              bg="#241C15"
              border="1px solid #241C15"
              borderRadius="50%"
              cursor="pointer"
              height="13px"
              marginLeft="-31px"
              width="13px"
            >
              <div
                className="sc-bdfBwQ iTpKPV"
                cursor="pointer"
                height="13px"
                width="13px"
              />
            </styled.div>
            <BodyCopy
              cursor="pointer"
              lineHeight="initial"
              margin={0}
              marginLeft="3em"
              width="234px"
            >
              <DynamicTextComponent
                cursor="pointer"
                fontFamily="IBM Plex Sans, Helvetica Neue, Arial, sans-serif"
                fontSize="1rem"
                fontWeight={400}
                letterSpacing={0}
                lineHeight="initial"
                margin={0}
                marginLeft="3em"
                tag="p"
                width="234px"
              >
                <styled.p
                  as="p"
                  cursor="pointer"
                  fontFamily="IBM Plex Sans, Helvetica Neue, Arial, sans-serif"
                  fontSize="1rem"
                  fontWeight={400}
                  letterSpacing={0}
                  lineHeight="initial"
                  margin={0}
                  marginLeft="3em"
                  width="234px"
                >
                  <p
                    className="sc-gKsewC hBsiQg"
                    cursor="pointer"
                    fontFamily="IBM Plex Sans, Helvetica Neue, Arial, sans-serif"
                    fontSize="1rem"
                    fontWeight={400}
                    letterSpacing={0}
                    width="234px"
                  >
                    dummy concept
                  </p>
                </styled.p>
              </DynamicTextComponent>
            </BodyCopy>
          </div>
        </styled.div>
      </li>
    </styled.li>
  </ProgressionListItem>
</Provider>
`;

exports[`ProgressionListItem should match previous snapshot for progression within a single module 1`] = `
<Provider
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <ProgressionListItem
    completion={67}
    concept={
      Object {
        "id": "qwer",
        "moduleId": "abcd",
        "title": "dummy concept",
      }
    }
    subject={
      Object {
        "description": "dummy description",
        "id": "zxcv",
        "modules": Array [
          Object {
            "concepts": Array [
              Object {
                "id": "qwer",
                "moduleId": "abcd",
                "order": 0,
                "title": "dummy concept",
              },
              Object {
                "id": "asdf",
                "moduleId": "abcd",
                "order": 1,
                "title": "dummy concept",
              },
            ],
            "description": "dummy description",
            "id": "abcd",
            "subjectId": "zxcv",
            "title": "dummy module",
          },
        ],
        "title": "dummy subject",
      }
    }
    updatedAt="2021-06-20T09:33:20.626274"
  >
    <styled.li
      borderLeft="1px solid #241C15"
      cursor="pointer"
      onClick={[Function]}
      paddingLeft="1.5em"
    >
      <li
        className="sc-hKgILt hnwkoi"
        cursor="pointer"
        onClick={[Function]}
      >
        <styled.div
          alignItems="center"
          cursor="pointer"
          display="flex"
          flexDirection="row"
        >
          <div
            className="sc-bdfBwQ gAYIqo"
            cursor="pointer"
            display="flex"
          >
            <styled.div
              bg="#241C15"
              border="1px solid #241C15"
              borderRadius="50%"
              cursor="pointer"
              height="13px"
              marginLeft="-31px"
              width="13px"
            >
              <div
                className="sc-bdfBwQ iTpKPV"
                cursor="pointer"
                height="13px"
                width="13px"
              />
            </styled.div>
            <styled.div
              cursor="pointer"
              marginLeft="1em"
              width="266px"
            >
              <div
                className="sc-bdfBwQ jSOydB"
                cursor="pointer"
                width="266px"
              >
                <GreatPrimer
                  cursor="pointer"
                  lineHeight="initial"
                  margin={0}
                >
                  <DynamicTextComponent
                    cursor="pointer"
                    fontFamily="IBM Plex Sans, Helvetica Neue, Arial, sans-serif"
                    fontSize="1.222rem"
                    fontWeight={400}
                    letterSpacing={0}
                    lineHeight="initial"
                    margin={0}
                    tag="p"
                  >
                    <styled.p
                      as="p"
                      cursor="pointer"
                      fontFamily="IBM Plex Sans, Helvetica Neue, Arial, sans-serif"
                      fontSize="1.222rem"
                      fontWeight={400}
                      letterSpacing={0}
                      lineHeight="initial"
                      margin={0}
                    >
                      <p
                        className="sc-gKsewC ebSZFE"
                        cursor="pointer"
                        fontFamily="IBM Plex Sans, Helvetica Neue, Arial, sans-serif"
                        fontSize="1.222rem"
                        fontWeight={400}
                        letterSpacing={0}
                      >
                        home.pages.auth.date
                      </p>
                    </styled.p>
                  </DynamicTextComponent>
                </GreatPrimer>
                <BodyCopy
                  cursor="pointer"
                  lineHeight="initial"
                  margin={0}
                >
                  <DynamicTextComponent
                    cursor="pointer"
                    fontFamily="IBM Plex Sans, Helvetica Neue, Arial, sans-serif"
                    fontSize="1rem"
                    fontWeight={400}
                    letterSpacing={0}
                    lineHeight="initial"
                    margin={0}
                    tag="p"
                  >
                    <styled.p
                      as="p"
                      cursor="pointer"
                      fontFamily="IBM Plex Sans, Helvetica Neue, Arial, sans-serif"
                      fontSize="1rem"
                      fontWeight={400}
                      letterSpacing={0}
                      lineHeight="initial"
                      margin={0}
                    >
                      <p
                        className="sc-gKsewC cIUUkE"
                        cursor="pointer"
                        fontFamily="IBM Plex Sans, Helvetica Neue, Arial, sans-serif"
                        fontSize="1rem"
                        fontWeight={400}
                        letterSpacing={0}
                      >
                        home.pages.auth.subject_completion
                      </p>
                    </styled.p>
                  </DynamicTextComponent>
                </BodyCopy>
              </div>
            </styled.div>
          </div>
        </styled.div>
        <styled.div
          alignItems="center"
          cursor="pointer"
          display="flex"
          flexDirection="row"
        >
          <div
            className="sc-bdfBwQ gAYIqo"
            cursor="pointer"
            display="flex"
          >
            <styled.div
              bg="#241C15"
              border="1px solid #241C15"
              borderRadius="50%"
              cursor="pointer"
              height="13px"
              marginLeft="-31px"
              width="13px"
            >
              <div
                className="sc-bdfBwQ iTpKPV"
                cursor="pointer"
                height="13px"
                width="13px"
              />
            </styled.div>
            <BodyCopy
              cursor="pointer"
              lineHeight="initial"
              margin={0}
              marginLeft="3em"
              width="234px"
            >
              <DynamicTextComponent
                cursor="pointer"
                fontFamily="IBM Plex Sans, Helvetica Neue, Arial, sans-serif"
                fontSize="1rem"
                fontWeight={400}
                letterSpacing={0}
                lineHeight="initial"
                margin={0}
                marginLeft="3em"
                tag="p"
                width="234px"
              >
                <styled.p
                  as="p"
                  cursor="pointer"
                  fontFamily="IBM Plex Sans, Helvetica Neue, Arial, sans-serif"
                  fontSize="1rem"
                  fontWeight={400}
                  letterSpacing={0}
                  lineHeight="initial"
                  margin={0}
                  marginLeft="3em"
                  width="234px"
                >
                  <p
                    className="sc-gKsewC hBsiQg"
                    cursor="pointer"
                    fontFamily="IBM Plex Sans, Helvetica Neue, Arial, sans-serif"
                    fontSize="1rem"
                    fontWeight={400}
                    letterSpacing={0}
                    width="234px"
                  >
                    dummy concept
                  </p>
                </styled.p>
              </DynamicTextComponent>
            </BodyCopy>
          </div>
        </styled.div>
      </li>
    </styled.li>
  </ProgressionListItem>
</Provider>
`;

exports[`ProgressionListItem should match previous snapshot for progression within the last concept of the last module 1`] = `
<Provider
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <ProgressionListItem
    completion={67}
    concept={
      Object {
        "id": "asdf",
        "moduleId": "abcd",
        "title": "dummy concept",
      }
    }
    subject={
      Object {
        "description": "dummy description",
        "id": "zxcv",
        "modules": Array [
          Object {
            "concepts": Array [
              Object {
                "id": "qwer",
                "moduleId": "abcd",
                "order": 0,
                "title": "dummy concept",
              },
              Object {
                "id": "asdf",
                "moduleId": "abcd",
                "order": 1,
                "title": "dummy concept",
              },
            ],
            "description": "dummy description",
            "id": "abcd",
            "subjectId": "zxcv",
            "title": "dummy module",
          },
        ],
        "title": "dummy subject",
      }
    }
    updatedAt="2021-06-20T09:33:20.626274"
  >
    <styled.li
      borderLeft="1px solid #241C15"
      cursor="pointer"
      onClick={[Function]}
      paddingLeft="1.5em"
    >
      <li
        className="sc-hKgILt hnwkoi"
        cursor="pointer"
        onClick={[Function]}
      >
        <styled.div
          alignItems="center"
          cursor="pointer"
          display="flex"
          flexDirection="row"
        >
          <div
            className="sc-bdfBwQ gAYIqo"
            cursor="pointer"
            display="flex"
          >
            <styled.div
              bg="#241C15"
              border="1px solid #241C15"
              borderRadius="50%"
              cursor="pointer"
              height="13px"
              marginLeft="-31px"
              width="13px"
            >
              <div
                className="sc-bdfBwQ iTpKPV"
                cursor="pointer"
                height="13px"
                width="13px"
              />
            </styled.div>
            <styled.div
              cursor="pointer"
              marginLeft="1em"
              width="266px"
            >
              <div
                className="sc-bdfBwQ jSOydB"
                cursor="pointer"
                width="266px"
              >
                <GreatPrimer
                  cursor="pointer"
                  lineHeight="initial"
                  margin={0}
                >
                  <DynamicTextComponent
                    cursor="pointer"
                    fontFamily="IBM Plex Sans, Helvetica Neue, Arial, sans-serif"
                    fontSize="1.222rem"
                    fontWeight={400}
                    letterSpacing={0}
                    lineHeight="initial"
                    margin={0}
                    tag="p"
                  >
                    <styled.p
                      as="p"
                      cursor="pointer"
                      fontFamily="IBM Plex Sans, Helvetica Neue, Arial, sans-serif"
                      fontSize="1.222rem"
                      fontWeight={400}
                      letterSpacing={0}
                      lineHeight="initial"
                      margin={0}
                    >
                      <p
                        className="sc-gKsewC ebSZFE"
                        cursor="pointer"
                        fontFamily="IBM Plex Sans, Helvetica Neue, Arial, sans-serif"
                        fontSize="1.222rem"
                        fontWeight={400}
                        letterSpacing={0}
                      >
                        home.pages.auth.date
                      </p>
                    </styled.p>
                  </DynamicTextComponent>
                </GreatPrimer>
                <BodyCopy
                  cursor="pointer"
                  lineHeight="initial"
                  margin={0}
                >
                  <DynamicTextComponent
                    cursor="pointer"
                    fontFamily="IBM Plex Sans, Helvetica Neue, Arial, sans-serif"
                    fontSize="1rem"
                    fontWeight={400}
                    letterSpacing={0}
                    lineHeight="initial"
                    margin={0}
                    tag="p"
                  >
                    <styled.p
                      as="p"
                      cursor="pointer"
                      fontFamily="IBM Plex Sans, Helvetica Neue, Arial, sans-serif"
                      fontSize="1rem"
                      fontWeight={400}
                      letterSpacing={0}
                      lineHeight="initial"
                      margin={0}
                    >
                      <p
                        className="sc-gKsewC cIUUkE"
                        cursor="pointer"
                        fontFamily="IBM Plex Sans, Helvetica Neue, Arial, sans-serif"
                        fontSize="1rem"
                        fontWeight={400}
                        letterSpacing={0}
                      >
                        home.pages.auth.subject_completion
                      </p>
                    </styled.p>
                  </DynamicTextComponent>
                </BodyCopy>
              </div>
            </styled.div>
          </div>
        </styled.div>
        <styled.div
          alignItems="center"
          cursor="pointer"
          display="flex"
          flexDirection="row"
        >
          <div
            className="sc-bdfBwQ gAYIqo"
            cursor="pointer"
            display="flex"
          >
            <styled.div
              bg="#241C15"
              border="1px solid #241C15"
              borderRadius="50%"
              cursor="pointer"
              height="13px"
              marginLeft="-31px"
              width="13px"
            >
              <div
                className="sc-bdfBwQ iTpKPV"
                cursor="pointer"
                height="13px"
                width="13px"
              />
            </styled.div>
            <BodyCopy
              cursor="pointer"
              lineHeight="initial"
              margin={0}
              marginLeft="3em"
              width="234px"
            >
              <DynamicTextComponent
                cursor="pointer"
                fontFamily="IBM Plex Sans, Helvetica Neue, Arial, sans-serif"
                fontSize="1rem"
                fontWeight={400}
                letterSpacing={0}
                lineHeight="initial"
                margin={0}
                marginLeft="3em"
                tag="p"
                width="234px"
              >
                <styled.p
                  as="p"
                  cursor="pointer"
                  fontFamily="IBM Plex Sans, Helvetica Neue, Arial, sans-serif"
                  fontSize="1rem"
                  fontWeight={400}
                  letterSpacing={0}
                  lineHeight="initial"
                  margin={0}
                  marginLeft="3em"
                  width="234px"
                >
                  <p
                    className="sc-gKsewC hBsiQg"
                    cursor="pointer"
                    fontFamily="IBM Plex Sans, Helvetica Neue, Arial, sans-serif"
                    fontSize="1rem"
                    fontWeight={400}
                    letterSpacing={0}
                    width="234px"
                  >
                    dummy concept
                  </p>
                </styled.p>
              </DynamicTextComponent>
            </BodyCopy>
          </div>
        </styled.div>
      </li>
    </styled.li>
  </ProgressionListItem>
</Provider>
`;
